fedscale/core/aggregator.py

```diff
@@ -194,16 +196,18 @@ def executor_info_handler(self, executorId, info):
 
         if len(self.registered_executor_info) == len(self.executors):
 
-            clientId = 1
+            num_of_clients = 1
             logging.info(f"Loading {len(info['size'])} client traces ...")
             for _size in info['size']:
                 # since the worker rankId starts from 1, we also configure the initial dataId as 1
-                mapped_id = clientId%len(self.client_profiles) if len(self.client_profiles) > 0 else 1
+                mapped_id = num_of_clients%len(self.client_profiles) if len(self.client_profiles) > 0 else 1
                 systemProfile = self.client_profiles.get(mapped_id, {'computation': 1.0, 'communication':1.0})
+
+                clientId = str(num_of_clients) if self.experiment_mode == events.SIMULATION_MODE else executorId
```

Comment:
clientId here is str type, it will cause a problem when you call select_dataset, which requires clientId to be int type.

Reply:
Fixed.